src/app/clusters/color-control-server/color-control-server.cpp

```diff
@@ -789,29 +816,34 @@ bool ColorControlServer::moveHueCommand(app::CommandHandler * commandObj, const
                                         uint8_t moveMode, uint16_t rate, uint8_t optionsMask, uint8_t optionsOverride,
                                         bool isEnhanced)
 {
-    EndpointId endpoint = commandPath.mEndpointId;
-
-    uint8_t currentHue          = 0;
-    uint16_t currentEnhancedHue = 0;
-    Status status               = Status::Success;
-
-    ColorHueTransitionState * colorHueTransitionState        = getColorHueTransitionState(endpoint);
-    Color16uTransitionState * colorSaturationTransitionState = getSaturationTransitionState(endpoint);
+    EndpointId endpoint                               = commandPath.mEndpointId;
+    Status status                                     = Status::Success;
+    ColorHueTransitionState * colorHueTransitionState = getColorHueTransitionState(endpoint);
 
     VerifyOrExit(colorHueTransitionState != nullptr, status = Status::UnsupportedEndpoint);
-    VerifyOrExit(colorSaturationTransitionState != nullptr, status = Status::UnsupportedEndpoint);
 
     if (!shouldExecuteIfOff(endpoint, optionsMask, optionsOverride))
     {
         commandObj->AddStatus(commandPath, Status::Success);
         return true;
     }
 
+    if (rate == 0 && moveMode != EMBER_ZCL_HUE_MOVE_MODE_STOP)
```

Comment:
Shouldn't this validity check be before the `shouldExecuteIfOff` check?  I think it should be, just like it is for MoveSaturation.

And a followup: When we later discover that the moveMode is invalid, InvalidAction does not look like the right error response.  It should be InvalidCommand or ConstraintError...

Reply:
done